src/OpenLoco/Map/MapGenerator.cpp

```diff
@@ -410,9 +411,20 @@ namespace OpenLoco::Map::MapGenerator
     // 0x004C4BD7
     static void generateWater(HeightMap& heightMap)
     {
-        _heightMap = heightMap.data();
-        call(0x004C4BD7);
-        _heightMap = nullptr;
+        static loco_global<uint16_t, 0x00525FB2> seaLevel;
+
+        Map::tile_loop tileLoop;
+        for (uint32_t posId = 0; posId < map_size; posId++)
+        {
+            auto pos = tileLoop.current();
+            auto tile = TileManager::get(pos);
+            auto* surface = tile.surface();
+
+            if (surface != nullptr && surface->baseZ() < (seaLevel << 2))
```

Comment:
What is the unit of sea level? Should we use a utility to function to convert it to tile Z?

Reply:
I don't know what the unit is, but it is likely similar to what OpenRCT2 is using (if not identical). Ideally, we find a good name for it and declare comparison operators for the type.

Reply:
I think OpenRCT2 multiplied by 8 rather than 4.

Reply:
That's what I remember as well, but Locomotion is using `shl bx, 2`. I wonder why it's different.